### DISCUSSION TITLE
Custom Carrier Board (Jetson AGX Orin) - Drivers Missing

### DISCUSSION THREAD

**Discussion by spb06** [OE4T/meta-tegra#1667] [Q&A] [answered]
Hey there, 

First off, thanks for providing support that this community has offered, it is a big help in bring up. 

### Background

We are working on bring up a custom carrier board that has the Jetson AGX Orin chip. Essentially, we used the Jetson AGX Orin Devkit's machine conf as a starting point and replaced the devkit's device tree with the custom carrier board's. We are working off Kirkstone. 

### Problem

The board boots up and I can verify that the custom carrier board's device tree has indeed been ported properly. However, certain drivers are missing and keeping the peripherals from being usable. 

For example, the custom carrier board has a 2.5G ethernet port that uses the `igc` driver (Intel® Ethernet Controller I225-IT) inside Linux Kernel. This peripheral does not show up when running `ifconfig -a`. Also, running `modinfo igc` results in `modinfo : ERROR : Module igc not found.' 

### Question

Do I have to go through any extra steps to pull in these drivers or should they be pulled in already based on the fact that they are listed in the device tree (device compatible string)?

**Comment by dwalkes** [comment]
Hi,

>  I have to go through any extra steps to pull in these drivers or should they be pulled in already based on the fact that they are listed in the device tree (device compatible string)

I believe you'll need to ensure they are included in the kernel yourself separately from the device tree config.  If you haven't done this previously the instructions at https://docs.yoctoproject.org/kernel-dev/common.html#configuring-the-kernel are pretty useful.

**Reply by dwalkes** [reply]
I should mention that since we don't use the upstream yocto kernel you'll substitute references to `linux-yocto` in the instructions I linked above with the kernel recipe in https://github.com/OE4T/meta-tegra/tree/master/recipes-kernel/linux depending on which branch you are targeting.  For [kirkstone](https://github.com/OE4T/meta-tegra/tree/kirkstone/recipes-kernel/linux) this will be `linux-tegra`.

**Reply by spb06** [reply, marked as answer]
Thanks for your help! I had to enable `igc` driver in the kernel using `bitbake -c menuconfig linux/kernel`.
